pkg/collector/kube.go

```diff
@@ -0,0 +1,39 @@
+package collector
+
+import (
+	"fmt"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type kubeCollector struct {
+	discoveryClient discovery.DiscoveryInterface
+}
+
+func newKubeCollector(kubeconfig string, discoveryClient discovery.DiscoveryInterface) (*kubeCollector, error) {
+	col := &kubeCollector{}
+
+	if discoveryClient == nil {
+		config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+		if err != nil {
+			return nil, err
+		}
+
+		col.discoveryClient, err = discovery.NewDiscoveryClientForConfig(config)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		col.discoveryClient = discoveryClient
+	}
+
+	return col, nil
+}
+
+func (c *kubeCollector) GetServerVersion() (string, error) {
+	versn, err := c.discoveryClient.ServerVersion()
```

Comment:
version!

Reply:
Fingers hurt... too many characters to type... done.